tfhe/src/js_on_wasm_api/js_high_level_api/zk.rs

```diff
@@ -111,7 +103,70 @@ impl CompactPkeCrs {
     }
 
     #[wasm_bindgen]
-    pub fn public_params(&self) -> CompactPkePublicParams {
-        CompactPkePublicParams(self.0.public_params().clone())
+    pub fn from_public_params(
+        public_params: &CompactPkePublicParams,
+    ) -> Result<CompactPkeCrs, JsError> {
+        catch_panic(|| {
+            CompactPkeCrs(crate::core_crypto::entities::CompactPkeCrs::from(
+                public_params.0.clone(),
+            ))
+        })
+    }
+}
+
+#[wasm_bindgen]
+pub struct CompactPkePublicParams(pub(crate) crate::zk::CompactPkePublicParams);
+
+// "wasm bindgen is fragile and prefers the actual type vs. Self"
+#[allow(clippy::use_self)]
+#[wasm_bindgen]
+impl CompactPkePublicParams {
```

Comment:
do we want to keep this on the js side ?

Reply:
I think we at least need to keep the deserialize code for backward compatibility ?

Reply:
ah is it only deserialization ?

Reply:
could it be put on th Crs and a "deserialize_from_parameters" kind of thing ?

Reply:
you want to keep both ? Maybe it's better to keep it simple and have only the Crs::deserialize_from_parameters ?

Reply:
sure, I'm just trying to think of users manipulating the old struct and not finding any function on it